src/database/common.c

```diff
@@ -419,6 +408,36 @@ void db_init(void)
 		dbversion = db_get_int(db, DB_VERSION);
 	}
 
+	// Update to version 10 if lower
+	if(dbversion < 10)
+	{
+		// Update to version 10: Use linking tables for queries table
+		logg("Updating long-term database to version 10");
+		if(!optimize_queries_table(db))
+		{
+			logg("Queries table not optimized, database not available");
+			dbclose(&db);
+			return;
+		}
+		// Get updated version
+		dbversion = db_get_int(db, DB_VERSION);
+	}
+
+	// Update to version 11 if lower
+	if(dbversion < 11)
```

Comment:
Do we need a two step upgrade here? It's all done in one PR so both could be done by upgrading to version 10.

Reply:
Some already checked out version 10 which modified their database. If I'd have squashed the other updated into this, they would have ended in an unsupported database.